app/models/user.rb

```diff
@@ -14,6 +14,12 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  # Returns the hash digest of the given string.
+  def User.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+    BCrypt :: Password.create string, cost: cost
```

Comment:
```suggestion
    BCrypt::Password.create string, cost: cost
```
cái này để làm gì e nhỉ

Reply:
Cái hàm này để băm một cái đoạn mã mình nhập xuống ạ, nó sẽ trả về 1 BCrypt::Password instace. Cost là option không bắt buộc, đại loại nó là chi phí bỏ ra để băm cái đoạn mã của mình, chi phí càng cao thì kết quả băm càng khó, càng bảo mật nhưng sẽ tăng thêm thời gian để giải mã ạ. Ở mỗi lần băm dù cùng 1 đầu vào nhưng nó sẽ băm ra những kết quả luôn luôn khác nhau :D Nên là đố hacker làm được gì ạ